lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -334,24 +334,67 @@ static bool isBinaryOp(linalg::GenericOp linalgOp) {
   return false;
 }
 
-// Return true if the operands are all the same.
-static bool allOperandsHaveSameType(linalg::GenericOp linalgOp) {
-  Type operandType = linalgOp->getOperand(0).getType();
-  for (Value operand : linalgOp->getOperands()) {
-    Type currentType = operand.getType();
-    if (currentType != operandType)
+// Returns true if all the operands of the current linalgOp are shape type and
+// have the same statically-known shape. For example: A: memref<56x32xf32,
+// strided<[32, 1], offset: ?>> B: memref<56x32xf32> allOperandsHaveSameShape(A,
+// B) return true. C: memref<1x32xf32> allOperandsHaveSameShape(A, C) return
+// false.
+bool allOperandsHaveSameShapeAndStrides(TypeRange types) {
+  assert(types.size() > 1 && "expect one or more types");
+  if (!types[0].isa<MemRefType>())
+    return false;
+  auto firstOperandType = types[0].cast<MemRefType>();
+
+  // Step1. Validate rank.
+  int64_t rankOperand = firstOperandType.getRank();
+  for (Type currentType : types) {
+    if (!currentType.isa<MemRefType>())
+      return false;
+    if (currentType.cast<MemRefType>().getRank() != rankOperand)
+      return false;
+  }
+
+  // Step2. Validate shape, and strides.
+  // Get stride and offset for the first operand.
+  int64_t offsetFirstOperand = 0;
+  SmallVector<int64_t> stridesFirstOperand;
+  if (failed(getStridesAndOffset(firstOperandType, stridesFirstOperand,
+                                 offsetFirstOperand)))
+    return false;
+  ArrayRef<int64_t> shapeOperand = firstOperandType.getShape();
```

Comment:
Nit: `shapeFirstOperand`?

Also, perhaps end with `Op` instead, to reduce line length?